enterprise/internal/campaigns/syncer.go

```diff
@@ -13,37 +14,226 @@ import (
 	"github.com/sourcegraph/sourcegraph/internal/httpcli"
 )
 
+// SyncRegistry manages a ChangesetSyncer per external service
+type SyncRegistry struct {
+	Parent      context.Context
+	Store       SyncStore
+	ReposStore  repos.Store
+	HTTPFactory *httpcli.Factory
+
+	priorityNotify chan []int64
+
+	mu      sync.Mutex
+	syncers map[int64]*ChangesetSyncer
+}
+
+// NewSycnRegistry creates a new sync registry which starts a syncer for each external service and will update them
+// when external services are changed, added or removed.
+func NewSyncRegistry(ctx context.Context, store SyncStore, repoStore repos.Store, cf *httpcli.Factory) *SyncRegistry {
+	r := &SyncRegistry{
+		Parent:         ctx,
+		Store:          store,
+		ReposStore:     repoStore,
+		HTTPFactory:    cf,
+		priorityNotify: make(chan []int64, 500),
+		syncers:        make(map[int64]*ChangesetSyncer),
+	}
+
+	services, err := repoStore.ListExternalServices(ctx, repos.StoreListExternalServicesArgs{})
+	if err != nil {
+		log15.Error("Fetching initial external services", "err", err)
+	}
+
+	// Add and start syncers
+	for _, service := range services {
+		r.Add(service.ID)
+	}
+
+	go r.handlePriorityItems()
+
+	return r
+}
+
+// Add adds a syncer for the supplied external service if it hasn't already been added and starts it.
```

Comment:
```suggestion
// Add adds a syncer for the supplied external service if it hasn't already been added and started.
```

BTW, what does "it" refers to? The registry or the syncer?

Reply:
It is the syncer, I'll update the comment